test/jasmine/tests/hover_label_test.js

```diff
@@ -132,6 +132,46 @@ describe('hover info', function() {
         });
     });
 
+    describe('hover info text with 0', function() {
+        var mockCopy = Lib.extendDeep({}, mock);
+
+        mockCopy.data[0].text = [];
+        // we treat number 0 as valid text
+        // see https://github.com/plotly/plotly.js/issues/2660
+        mockCopy.data[0].text[17] = 0;
+        mockCopy.data[0].hoverinfo = 'text';
+        mockCopy.data[0].mode = 'lines+markers+text';
+
+        beforeEach(function(done) {
+            Plotly.plot(createGraphDiv(), mockCopy.data, mockCopy.layout).then(done);
+        });
+
+        it('responds to hover text', function() {
+            var gd = document.getElementById('graph');
+            Fx.hover('graph', evt, 'xy');
+
+            var hoverTrace = gd._hoverdata[0];
+
+            expect(hoverTrace.curveNumber).toEqual(0);
+            expect(hoverTrace.pointNumber).toEqual(17);
+            expect(hoverTrace.x).toEqual(0.388);
+            expect(hoverTrace.y).toEqual(1);
+            expect(hoverTrace.text).toEqual(0);
```

Comment:
I see you're faithfully mimicking the previous test, but I'd prefer `toBe` instead of `toEqual` - lately I use `toBe` whenever possible, which I think includes all of these cases - especially for the `0`, since some values we would find unacceptable would pass, like `'' == 0`, whereas `toBe` is stricter.